rust/private/clippy.bzl

```diff
@@ -106,25 +122,38 @@ def _clippy_aspect_impl(target, ctx):
     if crate_info.is_test:
         args.rustc_flags.add("--test")
 
+    clippy_flags = None
+    if hasattr(ctx.attr, "_clippy_flags"):
+        clippy_flags = ctx.attr._clippy_flags[ClippyFlagsInfo].clippy_flags
+
     # For remote execution purposes, the clippy_out file must be a sibling of crate_info.output
     # or rustc may fail to create intermediate output files because the directory does not exist.
     if ctx.attr._capture_output[CaptureClippyOutputInfo].capture_output:
         clippy_out = ctx.actions.declare_file(ctx.label.name + ".clippy.out", sibling = crate_info.output)
         args.process_wrapper_flags.add("--stderr-file", clippy_out.path)
 
-        # If we are capturing the output, we want the build system to be able to keep going
-        # and consume the output. Some clippy lints are denials, so we treat them as warnings.
-        args.rustc_flags.add("-Wclippy::all")
+        if clippy_flags:
```

Comment:
Hmmm, feels like `capture_clippy_output` might also want to capture the success or failure of the action. I *think* I'd expect users to be able to pass `-D` and even in cases where a defect was found, the action doesn't fail and instead the output would represent the failure? What're your thoughts there? I'd kinda like to avoid dictating what flags users can pass at this level.

Reply:
I agree that that's the proper approach. It would however compicate things - we'd need to make changes to the process wrapper to capture the exit code of the clippy action, and somehow determine whether the non-0 exit code happened due to -D or something else (I haven't inspected clippy's exit codes) in order to know whether the action was successful.

I would like to avoid the current situation where a -D can exist for a while and the clippy action breaks due to a source code change when what it should have done instead is write to the file; that's why I think without changes to process wrapper we shouldn't allow -D. We could also step back and allow at most one of `--@rules_rust//:capture_clippy_output=clippy_flags=true` and `--@rules_rust//:capture_clippy_output=true`. That is IMO not ideal, but will allow this PR to progress.

Reply:
I added a check that fail the build if both --@rules_rust//:capture_clippy_output=true and non-empty ---@rules_rust//:clippy_flags are specified.

Reply:
> I added a check that fail the build if both --@rules_rust//:capture_clippy_output=true and non-empty ---@rules_rust//:clippy_flags are specified.

@scentini, is this that you had in mind?

Reply:
Yes, this looks good to me, thanks!

Reply:
Using `--cap-lints warn` instead of `-W clippy:all` allows us to guarantee that the build doesn't fail from denied lints while also allowing the user to specify their own clippy flags. I've implemented this in #2451.

(It still means that denied lints show up as warnings in the captured output, but that's no different from `-W clippy::all`)